python/code_challenges/linked_list_zip.py

```diff
@@ -1,2 +1,20 @@
+from data_structures.linked_list import LinkedList
+
 def zip_lists(a, b):
-    pass
+    current_a = a.head
+    current_b = b.head
+
+    while current_b and current_a:
+            a.insert_after(current_a.value, current_b.value)
```

Comment:
Starting at line 95 in linked_list.py 

```
        current = self.head
        while current:
            if current.value is value:
                new_node.next = current.next
                current.next = new_node
                break
```
It looks like the value of current.value does not equal value causing the if block to fail. Since the if block exists in side of a while loop that's the cause of your infinite loop.